controllers/config_helper_test.go

```diff
@@ -0,0 +1,274 @@
+				Name:      "config",
+			},
+			Spec: v1alpha1.ConfigSpec{},
+		}
+
+		gatekeeper = &operatorv1alpha1.Gatekeeper{
+			ObjectMeta: metav1.ObjectMeta{
+				UID: "test-1",
+			},
+			Spec: operatorv1alpha1.GatekeeperSpec{
+				Config: &operatorv1alpha1.ConfigConfig{
+					Matches: []v1alpha1.MatchEntry{{
+						ExcludedNamespaces: []wildcard.Wildcard{
+							"cat-ns", "dog-ns",
+						},
+						Processes: []string{
+							"audit", "webhook", "sync",
+						},
+					}},
+				},
+			},
+		}
+
+		setExemptNamespaces(ctx, c, config, gatekeeper, scheme, Log)
+		g.Expect(config.Spec.Match).Should(HaveLen(2))
+		g.Expect(config.Spec.Match[0]).Should(BeComparableTo(defaultConfig.Spec.Match[0]))
+		g.Expect(config.Spec.Match[1]).Should(BeComparableTo(gatekeeper.Spec.Config.Matches[0]))
+	})
+
+	t.Run("Should be same result when execute setExemptNamespaces several times ", func(t *testing.T) {
+		config = &v1alpha1.Config{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace,
+				Name:      "config",
+			},
+			Spec: v1alpha1.ConfigSpec{},
+		}
+
+		disableDefaultMatches = false
+
+		gatekeeper = &operatorv1alpha1.Gatekeeper{
+			ObjectMeta: metav1.ObjectMeta{
+				UID: "test-1",
+			},
+			Spec: operatorv1alpha1.GatekeeperSpec{
+				Config: &operatorv1alpha1.ConfigConfig{
+					Matches: []v1alpha1.MatchEntry{{
+						ExcludedNamespaces: []wildcard.Wildcard{
+							"cat-ns", "dog-ns",
+						},
+						Processes: []string{
+							"audit", "webhook", "sync",
+						},
+					}},
+					DisableDefaultMatches: disableDefaultMatches,
+				},
+			},
+		}
+		// First try
+		setExemptNamespaces(ctx, c, config, gatekeeper, scheme, Log)
+
+		// Second try
+		setExemptNamespaces(ctx, c, config, gatekeeper, scheme, Log)
+
+		// Third try
+		setExemptNamespaces(ctx, c, config, gatekeeper, scheme, Log)
+		g.Expect(config.Spec.Match).Should(HaveLen(2))
+		g.Expect(config.Spec.Match[0]).Should(BeComparableTo(defaultConfig.Spec.Match[0]))
+		g.Expect(config.Spec.Match[1]).Should(BeComparableTo(gatekeeper.Spec.Config.Matches[0]))
+	})
+
+	t.Run("Should not change config when ownerRef is different and match exist", func(t *testing.T) {
+		config = &v1alpha1.Config{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace,
+				Name:      "config",
+			},
+			Spec: v1alpha1.ConfigSpec{
+				Match: []v1alpha1.MatchEntry{{
+					ExcludedNamespaces: []wildcard.Wildcard{
+						"cat-ns", "dog-ns",
+					},
+					Processes: []string{
+						"audit", "webhook", "sync",
+					},
+				}},
+			},
+		}
+
+		disableDefaultMatches = false
+
+		gatekeeper = &operatorv1alpha1.Gatekeeper{
+			ObjectMeta: metav1.ObjectMeta{
+				UID: "test-22",
+			},
+			Spec: operatorv1alpha1.GatekeeperSpec{
+				Config: &operatorv1alpha1.ConfigConfig{
+					Matches: []v1alpha1.MatchEntry{{
+						ExcludedNamespaces: []wildcard.Wildcard{
+							"tiger-ns", "rabbit-ns",
+						},
+						Processes: []string{
+							"audit",
+						},
+					}},
+				},
+			},
+		}
+		// First try
```

Comment:
```suggestion
```

Reply:
I found it confusing because `first try` indicates there should be more than one try and there wasn't in that test.

Reply:
oh it was line 268 lol sorry I thought it was line 218 Thanks!!